src/pages/SingleCell/HomePage.tsx

```diff
@@ -0,0 +1,1168 @@
+                                    id="entitySelect"
+                                    // className="custom-dropdown"
+                                    onChange={
+                                        this.handleEntitySelectChangeStack
+                                    }
+                                    value={
+                                        singleCellStore.stackEntity
+                                            ? singleCellStore.stackEntity
+                                            : ''
+                                    }
+                                    options={entityNames.map(entityName => ({
+                                        value: entityName,
+                                        label: entityName.replace(/_/g, ' '),
+                                    }))}
+                                    placeholder={
+                                        selectedOption &&
+                                        selectedOption.includes('type')
+                                            ? 'Sort by cell type...'
+                                            : selectedOption &&
+                                              selectedOption.includes('cycle')
+                                            ? 'Sort by cycle phase...'
+                                            : 'Sort by ...'
+                                    }
+                                    isDisabled={!selectedOption}
+                                    clearable={false}
+                                    searchable={true}
+                                />
+                            </div>
+                        )}
+
+                        {chartType === 'pie' && (
+                            <div className="checkbox-wrapper-3">
+                                <input
+                                    type="checkbox"
+                                    id="cbx-3"
+                                    checked={singleCellStore.tooltipEnabled}
+                                    onChange={this.handleTooltipCheckboxChange}
+                                />
+                                <label htmlFor="cbx-3" className="toggle">
+                                    <span></span>
+                                </label>
+                                <label
+                                    htmlFor="cbx-3"
+                                    className="toggle-label"
+                                    style={{
+                                        fontWeight: 'normal',
+                                        fontSize: '14px',
+                                        marginLeft: '10px',
+                                    }}
+                                >
+                                    Show the data table
+                                </label>
+                            </div>
+                        )}
+                        {chartType === 'stack' && (

```

Comment:
In the next lines, you are conditionally rendering the different div elements if  `chartType === 'stack'`. Do you need to repeat `chartType === 'stack'` three times? I think you can combine the conditional rendering.